src/p2p_networking.rs

```diff
@@ -0,0 +1,52 @@
+use crate::errors::RuntimeError;
+use crate::{AsyncRuntime, PeerManager};
+use bitcoin::secp256k1::PublicKey;
+use log::debug;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::task::Poll;
+use std::time::Duration;
+use tokio::time::sleep;
+
+pub fn connect_to_peer(
+    rt: &AsyncRuntime,
+    peer_manager: Arc<PeerManager>,
+    pubkey: PublicKey,
+    addr: SocketAddr,
+) -> Result<(), RuntimeError> {
+    rt.handle().block_on(async move {
+        let result =
+            lightning_net_tokio::connect_outbound(Arc::clone(&peer_manager), pubkey, addr).await;
+
+        if let Some(connection_closed_future) = result {
+            let mut connection_closed_future = Box::pin(connection_closed_future);
```

Comment:
Why do we need to pin here?

Reply:
I think it's because `poll(future)` [requires its param to implement the Unpin trait](https://github.com/rust-lang/futures-rs/blob/7a98cf0bbeb397dcfaf5f020b371ab9e836d33d4/futures-util/src/async_await/poll.rs#L24).

Reply:
https://rust-lang.github.io/async-book/04_pinning/01_chapter.html

Reply:
Thanks!